src/Massive.Shared.cs

```diff
@@ -1410,30 +1471,251 @@ public interface IConnectionStringProvider
+#else
+				ProviderFactory = DbProviderFactories.GetFactory(providerName);
+#endif
+			}
+		}
+	}
+#pragma warning restore 0618
+
+
+	/// <summary>
+	/// Massive equivalent of System.Data.Common.DbProviderFactories (which does not exist in .NET Core): a registry of supported providers
+	/// for provider names known to Massive.
+	/// To use providers not yet known to Massive, just pass your own implementation of ConnectionProvider to the DynamicModel constructor.
+	/// </summary>
+	/// <remarks>
+	/// This factory is always used by PureConnectionStringProvider (on .NET Core and on .NET Framework).
+	/// The old System.Data.Common.DbProviderFactories is used by ConfigFileConnectionProvider (.NET Framework only).
+	/// So: if you pass a pure connection string to Massive then it won't look for providers registered in the traditional way, not even on .NET Framework;
+	/// however if you pass a connection string name (the old approach, available on .NET Framework only) then everything will work as it used to.
+	/// </remarks>
+	public class DynamicModelProviderFactories
+	{
+		private const string INSTANCE_FIELD_NAME = "Instance";
+
+		/// <summary>
+		/// Return the factory object for the ADO.NET provider, for provider names known to Massive.
+		/// </summary>
+		/// <param name="providerName">The provider name</param>
 		/// <returns></returns>
-		public string GetProviderName(string connectionStringName)
+		public static DbProviderFactory GetFactory(string providerName)
+		{
+			string assemblyName = null;
+			var factoryClass = GetProviderFactoryClass(providerName);
+			string[] elements = factoryClass.Split(',');
+			string factoryClassName = elements[0];
+			if(elements.Length > 1)
+			{
+				assemblyName = elements[1];
+			}
+			else
+			{
+				assemblyName = assemblyName ?? factoryClassName.Substring(0, factoryClassName.LastIndexOf("."));
+			}
+			var assemblyNameClass = new AssemblyName(assemblyName);
+			Type type = Assembly.Load(assemblyNameClass).GetType(factoryClassName);
+			var f = type.GetField(INSTANCE_FIELD_NAME);
+			if(f == null)
+			{
+				throw new NotImplementedException("No " + INSTANCE_FIELD_NAME + " field/property found in intended DbProviderFactory class '" + factoryClassName + "'");
+			}
+			return (DbProviderFactory)f.GetValue(null);
+		}
+
+
+		/// <summary>
+		/// Get factory class name based on known provider name.
+		/// </summary>
+		/// <param name="providerName">Provider name</param>
+		/// <returns>"fully.qualified.provider.class, assembly.name", if assembly.name is ommitted then everything up to the class name is used instead</returns>
+		private static string GetProviderFactoryClass(string providerName)
 		{
-			var providerName = ConfigurationManager.ConnectionStrings[connectionStringName].ProviderName;
-			return !string.IsNullOrWhiteSpace(providerName) ? providerName : null;
+			switch(providerName.ToLowerInvariant())
+			{
+				case "system.data.sqlclient":
```

Comment:
This stuff can't be stored in Massive.Shared.cs. The factory names are stored with the db specific file, that's done for a reason. You now moved them here. that's not maintainable.